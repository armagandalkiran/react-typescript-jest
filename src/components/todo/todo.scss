@import "../../common.scss";

.todo {
  &__container {
    padding: 20px;
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
  }

  &__content {
    width: 60%;
    @include respond(phone) {
      width: 95%;
    }
  }

  &__error--message {
    width: 80%;
    font-size: 0.9rem;
    color: $error;
    @include respond(phone) {
      font-size: 0.7rem;
    }
  }

  &__list--container {
    width: 80%;
    padding: 10px;
    display: flex;
    flex-direction: column;
    @include respond(phone) {
      padding: 10px 0;
    }
  }
}
