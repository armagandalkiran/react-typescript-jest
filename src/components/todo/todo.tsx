import { useState } from "react";
import { defaultValues } from "../../models/constants";
import { TodoListItems } from "../todoListItems";
import { TextInput } from "../textInput";
import "./todo.scss";

export const ToDo = () => {
  const [text, setText] = useState("");
  const [tasks, setTasks] = useState(defaultValues);
  const [error, setError] = useState("");

  const displayError = () => {
    return error && <p className="todo__error--message">{error}</p>;
  };

  return (
    <div className="todo__container">
      <div className="todo__content">
        <TextInput
          text={text}
          setText={setText}
          setTasks={setTasks}
          setError={setError}
        />
        <div>{displayError()}</div>
        <ul className="todo__list--container">
          {tasks.map((data, idx) => (
            <TodoListItems
              key={idx}
              idx={idx}
              data={data}
              setTasks={setTasks}
              tasks={tasks}
            />
          ))}
        </ul>
      </div>
    </div>
  );
};
