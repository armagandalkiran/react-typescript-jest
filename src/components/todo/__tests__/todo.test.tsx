import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { ToDo } from "../todo";

describe("ToDo tests", () => {
  let button, input;

  beforeEach(() => {
    render(<ToDo />);
    button = screen.getByText("+");
    input = screen.getByTestId("input");
  });

  afterEach(() => {
    cleanup();
  });

  test("Test default tasks render", ()=>{
    const items = screen.getAllByText(/default/i);
    expect(items.length).toEqual(3)
  })

    test("matches snapshot", () => {
      const tree = renderer.create(<ToDo />).toJSON();
      expect(tree).toMatchSnapshot();
    });
});
