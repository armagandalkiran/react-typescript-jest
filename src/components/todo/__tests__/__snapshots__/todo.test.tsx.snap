// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDo tests matches snapshot 1`] = `
<div
  className="todo__container"
>
  <div
    className="todo__content"
  >
    <div
      className="todo__entry--wrapper"
    >
      <input
        className="todo__input"
        data-testid="input"
        onChange={[Function]}
        placeholder="Enter your task.."
        type="text"
        value=""
      />
      <button
        className="todo__button"
        data-testid="button"
        onClick={[Function]}
      >
        Ekle
      </button>
    </div>
    <div />
    <ul
      className="todo__list--container"
    >
      <li
        className="todo__list--item"
      >
        <p
          className="todo__task--checked"
        >
          default 1
        </p>
        <div
          className="todo__actions--container"
        >
          <input
            checked={true}
            className="todo__actions--checkbox"
            id="todoCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="todo__actions--deleteContainer"
          >
            <img
              alt="default 1"
              className="todo__actions--delete"
              onClick={[Function]}
              src="trash-icon.svg"
            />
          </div>
        </div>
      </li>
      <li
        className="todo__list--item"
      >
        <p
          className="todo__task--notChecked"
        >
          default 2
        </p>
        <div
          className="todo__actions--container"
        >
          <input
            checked={false}
            className="todo__actions--checkbox"
            id="todoCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="todo__actions--deleteContainer"
          >
            <img
              alt="default 2"
              className="todo__actions--delete"
              onClick={[Function]}
              src="trash-icon.svg"
            />
          </div>
        </div>
      </li>
      <li
        className="todo__list--item"
      >
        <p
          className="todo__task--notChecked"
        >
          default 3
        </p>
        <div
          className="todo__actions--container"
        >
          <input
            checked={false}
            className="todo__actions--checkbox"
            id="todoCheckbox"
            onChange={[Function]}
            type="checkbox"
          />
          <div
            className="todo__actions--deleteContainer"
          >
            <img
              alt="default 3"
              className="todo__actions--delete"
              onClick={[Function]}
              src="trash-icon.svg"
            />
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
